lib/runtime/text-encoder.js

```diff
@@ -0,0 +1,25 @@
+const textEncoding = require('text-encoding')
+
+function CustomTextDecoder () {
```

Comment:
I believe the TextEncoder constructor has optional params: https://github.com/inexorabletash/text-encoding/blob/master/lib/encoding.js#L984

Additionally, there are [properties](https://github.com/inexorabletash/text-encoding/blob/master/lib/encoding.js#L1051) on TextDecoder that should be defined on this CustomerTextDecoder. I want to avoid copying them from the implementation, though. Off the cuff, I'm not sure how we should do that. Maybe a [Proxy](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Proxy)?